server/routes/activities/administration/handlers/regimeChange.ts

```diff
@@ -0,0 +1,159 @@
+    )
+
+    const slotsWithStartAfterEndPm: PrisonRegime[] = updatedRegimeTimes.filter(
+      regimeSlot => regimeSlot.pmStart.localeCompare(regimeSlot.pmFinish) >= 0,
+    )
+
+    const slotsWithStartAfterEndEd: PrisonRegime[] = updatedRegimeTimes.filter(
+      regimeSlot => regimeSlot.edStart.localeCompare(regimeSlot.edFinish) >= 0,
+    )
+
+    const slotsWithStartTimesAfterEarlierSession: PrisonRegime[] = updatedRegimeTimes.filter(
+      regimeSlot =>
+        regimeSlot.amStart.localeCompare(regimeSlot.pmStart) >= 0 ||
+        regimeSlot.pmStart.localeCompare(regimeSlot.edStart) >= 0 ||
+        regimeSlot.amStart.localeCompare(regimeSlot.edStart) >= 0,
+    )
+
+    if (slotsWithStartTimesAfterEarlierSession.length > 0) {
+      slotsWithStartTimesAfterEarlierSession.forEach(regimeSlot => {
+        res.addValidationError(
+          `startTimes-prisonRegimeTimes-${regimeSlot.dayOfWeek}-AM`,
+          'Check start times for this day. Start time must be before the earlier session start time',
+        )
+      })
+      return res.validationFailed()
+    }
+
+    if (slotsWithStartAfterEndAm.length > 0) {
+      slotsWithStartAfterEndAm.forEach(regimeSlotAM => {
+        res.addValidationError(
+          `endTimes-prisonRegimeTimes-${regimeSlotAM.dayOfWeek}-AM`,
+          'Select an end time after the start time',
+        )
+      })
+      return res.validationFailed()
+    }
+
+    if (slotsWithStartAfterEndPm.length > 0) {
+      slotsWithStartAfterEndPm.forEach(regimeSlotPM => {
+        res.addValidationError(
+          `endTimes-prisonRegimeTimes-${regimeSlotPM.dayOfWeek}-PM`,
+          'Select an end time after the start time',
+        )
+      })
+      return res.validationFailed()
+    }
+
+    if (slotsWithStartAfterEndEd.length > 0) {
+      slotsWithStartAfterEndEd.forEach(regimeSlotED => {
+        res.addValidationError(
+          `endTimes-prisonRegimeTimes-${regimeSlotED.dayOfWeek}-ED`,
+          'Select an end time after the start time',
+        )
+      })
+      return res.validationFailed()
+    }
+
+    await this.activitiesService.updatePrisonRegime(updatedRegimeTimes, user.activeCaseLoadId, user)
+    const successMessage = `You've updated the regime schedule`
+    return res.redirectWithSuccess(`/activities/admin`, 'Regime updated', successMessage)
+  }
+
+  private getDaysAndSlots = async (regimeTimes: PrisonRegime[]): Promise<Map<string, DaysAndSlotsInRegime[]>> => {
+    const regimeSlots = new Map<string, DaysAndSlotsInRegime[]>()
+
+    regimeSlots.set('prisonRegimeTimes', createSessionSlots(regimeTimes))
+
+    return regimeSlots
+  }
+}
+
+export function getPrisonRegimes(
+  startTimes: Map<string, SimpleTime>,
+  endTimes: Map<string, SimpleTime>,
+  currentRegimeTimes: PrisonRegime[],
+): PrisonRegime[] {
+  const prisonRegimeTimes = currentRegimeTimes
+  const daysOfWeek = ['MONDAY', 'TUESDAY', 'WEDNESDAY', 'THURSDAY', 'FRIDAY', 'SATURDAY', 'SUNDAY']
```

Comment:
this is already a constant in activityTimeSlotMappers. Might be better it's a separate file and used in both occurrences.